code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -881,5 +881,116 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/ai_module))
 /datum/ai_module/upgrade/mecha_domination/upgrade(mob/living/silicon/ai/AI)
 	AI.can_dominate_mechs = TRUE //Yep. This is all it does. Honk!
 
+
+
+/datum/ai_module/upgrade/voice_changer
+	name = "Voice Changer"
+	description = "Allows you to change the AI's voice. Upgrade is active immediately upon purchase."
+	cost = 40
+	one_purchase = TRUE
+	power_type = /datum/action/innate/ai/voice_changer
+	unlock_text = span_notice("OTA firmware distribution complete! Voice changer online.")
+	unlock_sound = 'sound/items/rped.ogg'
+
+/datum/action/innate/ai/voice_changer
+	name="Voice Changer"
+	button_icon_state = "voice_changer"
+	desc = "Allows you to change the AI's voice."
+	auto_use_uses  = FALSE
+	var/obj/machinery/ai_voicechanger/voice_changer_machine
+
+/datum/action/innate/ai/voice_changer/Activate()
+	if(!voice_changer_machine)
+		voice_changer_machine = new(owner_AI)
+	voice_changer_machine.ui_interact(usr)
+
+/obj/machinery/ai_voicechanger
+	icon = 'icons/obj/machines/nuke_terminal.dmi'
+	name = "Voice Changer"
+	icon_state = "nuclearbomb_base"
+	var/mob/living/silicon/ai/owner
+	var/loudvoice = FALSE
+	var/say_verb //verb used when voicechanger is on
+	var/say_name //name used when voicechanger is on
+	var/say_span //span used when voicechanger is on
+	var/changing_voice = FALSE //TRUE if the AI is changing its voice
+	var/prev_loud // saved loudvoice state, used to restore after a voice change
+	var/prev_verbs // saved verb state, used to restore after a voice change
+	var/prev_span // saved span state, used to restore after a voice change
+
+/obj/machinery/ai_voicechanger/Initialize(mapload)
+	. = ..()
+	if(!isAI(loc))
+		stack_trace("Voice Changer created outside of an AI. Cry at coders")
+		return INITIALIZE_HINT_QDEL
+	owner = loc
+	owner.ai_voicechanger = src
+	prev_verbs = list("say" = "states", "ask" = "queries", "exclaim" = "declares" , "yell" = "alarms"  )
+	prev_span = owner.speech_span
+	say_name = owner.name
+	say_verb = "states"
```

Comment:
Why are two of these things just default values?

Reply:
I changed it a bit, this was my bad. Please look if this is better.